### ISSUE TITLE
Support for Gnome Shell 45 (make the extension an ESModule)

### ISSUE THREAD

**Issue by johnnynoone** [RensAlthuis/vertical-overview#128]
Just a heads-up: apparently [Gnome 45 will require extensions to be ESModules](https://news.itsfoss.com/gnome-45-extensions/).

**Comment by systemd11** [comment]
Would be great

**Comment by coldtobi** [comment]
Debian bug: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1052110 with some information like a link to a porting guide

**Comment by stdevPavelmc** [comment]
Yup, bring it up to Gnome 45 please.

**Comment by mtsoltan** [comment]
For everyone who is following this issue, my fork of this repository has Gnome Shell 45 support:
https://github.com/mtsoltan/vertical-overview

You can clone it into `git clone git@github.com:mtsoltan/vertical-overview.git ~/.local/share/gnome-shell/extensions/vertical-overview@mtsoltan.github.com` and it should work.

My changes were aimed at getting it to work, rather than being good practices, and therefore, I won't open a PR with them.
The extension settings window looks a bit weird, but is completely usable.

However, for anyone looking to use this extension with Gnome Shell  45, it should be your best bet until the official repo is updated.

Edit: If your key-binds for previous and next workspaces are not working, you have to change them from d-conf editor.

Go to `org.gnome.desktop.wm.keybindings` and try setting `move-to-workspace-up` and `move-to-workspace-down` instead of `move-to-workspace-left` and `move-to-workspace-right`.

**Comment by xuhcc** [comment]
@mtsoltan I tried your fork and can confirm that it works on GNOME 45. Thank you!

I think it is fine if you open a PR, because working code is better than nothing. Or just upload your version to extension.gnome.org. this repo looks abandoned anyway.

**Comment by RensAlthuis** [comment]
Yeah, I don't have time to work on this and I'm also no longer using it myself any longer so it is pretty much abandoned.
I am willing to hand over give permissions or transfer ownership of the repo if someone is willing to keep it updated.

**Comment by suwalski** [comment]
@mtsoltan Your port works quite well. The only issue I have is that it seems to crash after some time (possibly after screen lock) with "swipeTracker is not defined". It has the same error for me if I try to turn it off.

**Comment by lvanderree** [comment]
> @mtsoltan Your port works quite well. The only issue I have is that it seems to crash after some time (possibly after screen lock) with "swipeTracker is not defined". It has the same error for me if I try to turn it off.

I disabled (I think) three-finge-swipes  and since then haven't seen the issue anymore...
https://github.com/mtsoltan/vertical-overview/blob/master/gestures.js#L12
```
    if (USE_3_FINGER_SWIPES && false) {
```

**Comment by achirkin** [comment]
@mtsoltan your port works well for me as well, and I see the same error as @suwalski as well. It can be consistently reproduced by locking the screen and logging back in. Trace:
```
user@computer:~$ sudo journalctl /usr/bin/gnome-shell | grep 'vertical-overview'
Nov 20 10:57:29 computer gnome-shell[4450]: JS ERROR: Extension vertical-overview@mtsoltan.github.com: ReferenceError: swipeTracker is not defined
                                                   reset@file:///home/user/.local/share/gnome-shell/extensions/vertical-overview@mtsoltan.github.com/gestures.js:73:9
                                                   disable@file:///home/user/.local/share/gnome-shell/extensions/vertical-overview@mtsoltan.github.com/extension.js:52:18

```
I'm not familiar with ES. What's the scope of the `swipeTracker` in the [gestures.js/reset()](https://github.com/mtsoltan/vertical-overview/blob/24d21da415bc08d04782b174a40e391afbeb666b/gestures.js#L73)?
Shouldn't it be initialized within the function?
